firebase-config-interop/src/main/java/com/google/firebase/remoteconfig/interop/rollouts/RolloutsState.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.remoteconfig.interop.rollouts;
+
+import androidx.annotation.NonNull;
+import com.google.auto.value.AutoValue;
+import java.util.Set;
+
+/** Model representing the state of all rollouts assigned to an app instance at a point in time. */
+@AutoValue
+public abstract class RolloutsState {
+  /** Rollouts assigned at this point in time. May include zero or more assignments. */
+  @NonNull
+  public abstract Set<RolloutAssignment> getRolloutAssignments();
```

Comment:
Saw a warning in unit test, suggest using `ImmutableSet` https://github.com/firebase/firebase-android-sdk/actions/runs/5894721664/job/15989122819?pr=5255

Reply:
Good call, I see that's AutoValue's recommendation and likely what we'll want to pass from RC anyway. Updated.

Reply:
I noticed that none of the other Firebase SDKs depend on guava or use any of the `Immutable*` classes outside of tests. Just based on some quick digging around, it looks like there's a lot of concern about the size of the dependency. I'm going to keep `Set` here